llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -8424,6 +8424,12 @@ TargetLowering::createSelectForFMINNUM_FMAXNUM(SDNode *Node,
 
   if (Node->getFlags().hasNoNaNs()) {
     ISD::CondCode Pred = Opcode == ISD::FMINNUM ? ISD::SETLT : ISD::SETGT;
+    EVT VT = Node->getValueType(0);
+    if ((!isCondCodeLegal(Pred, VT.getSimpleVT()) ||
+         !isOperationLegalOrCustom(ISD::VSELECT, VT)) &&
+        VT.isVector() &&
+        isOperationLegal(Node->getOpcode(), VT.getScalarType()))
```

Comment:
Can you remove the check about the scalar type? The point was to check positively for the legality of the operations you are directly introducing, not the speculative operations that may be introduced later